beacon-chain/rpc/validator/aggregator.go

```diff
@@ -70,13 +70,26 @@ func (as *Server) SubmitAggregateSelectionProof(ctx context.Context, req *ethpb.
 			return nil, status.Errorf(codes.Internal, "Could not find attestation for slot and committee in pool")
 		}
 	}
+
+	var indexInCommittee uint64
+	for i, idx := range committee {
+		if idx == validatorIndex {
+			indexInCommittee = uint64(i)
+		}
+	}
+
 	best := aggregatedAtts[0]
 	for _, aggregatedAtt := range aggregatedAtts[1:] {
-		if aggregatedAtt.AggregationBits.Count() > best.AggregationBits.Count() {
+		// The aggregator should prefer an attestation that they have signed. We check this by
+		// looking at the attestation's committee index against the validator's committee index
+		// and check the aggregate bits to ensure the validator's index is set.
+		if aggregatedAtt.Data.CommitteeIndex == req.CommitteeIndex &&
```

Comment:
What happens here? Assume attestations are in the following order:
```
indexInCommittee=0
AggregationBits has a length 4
local aggregation bits is 0b0001
local head root is 0xA

First attestation: {HeadRoot: 0xA, AggregationBits: 0x0001} 
Second attestation: {HeadRoot: 0xB, AggregationBits: 0x1100}  // Best
```

Wouldn't second attestation overrides the first attestation because it has more bits? We want the first attestation to be test

Reply:
The best is only replaced by an attestation where where 0b0001 is set, so the first attestation is considered the best.

Reply:
When the second attestation comes, wouldn't you get:
`aggregatedAtt.Data.CommitteeIndex == req.CommitteeIndex` true
`aggregatedAtt.AggregationBits.BitAt(indexInCommittee)` false
`!best.AggregationBits.BitAt(indexInCommittee)` false
`aggregatedAtt.AggregationBits.Count() > best.AggregationBits.Count())` true

`true && false && false || true` equals to  true?

Reply:
It’s `true && false && (false || true)`

Reply:
Ah I missed the (), note to self dont review code sunday late. 

What about this case?
```
local aggregation bits is 0b0001
local head root is 0xA

First attestation: {HeadRoot: 0xB, AggregationBits: 0x0010} 
Second attestation: {HeadRoot: 0xC, AggregationBits: 0x1100}  
```

Second attestation should be the best one but I don't think it'd be replaced given this condition?
`true && false && (false || true)`

Reply:
Yeah that is correct. I figured that the aggregator would almost always have their attestation present, but let me see about this.

Reply:
OK this has been added. Please review https://github.com/prysmaticlabs/prysm/pull/6566/commits/8827cb1f45cb8be4f62e62654da70fe83e5a7e37